src/execution_plan/execution_plan.c

```diff
@@ -568,7 +568,10 @@ static void _buildMergeMatchStream(ExecutionPlan *plan, const cypher_astnode_t *
 	const cypher_astnode_t *path = cypher_ast_merge_get_pattern_path(clause);
 	AST *rhs_ast = AST_MockMatchPattern(ast, path);
 
+	// Populate sub-ExecutionPlan.
 	_PopulateExecutionPlan(rhs_plan, NULL);
+	// Add filter ops to sub-ExecutionPlan.
+	if(rhs_plan->filter_tree) _ExecutionPlan_PlaceFilterOps(rhs_plan, NULL);
```

Comment:
It is kind of a misfortune that we need to call `_ExecutionPlan_PlaceFilterOps` from different places, I wish there could be a single place from which we call this function